adafruit_lsm6ds.py

```diff
@@ -163,74 +165,73 @@ class LSM6DSOX: #pylint: disable=too-many-instance-attributes
+    _ois_pu_dis = RWBit(_LSM6DS_PIN_CTRL, 7)
+    _spi2_read_en = RWBit(_LSM6DS_UI_INT_OIS, 3)
+    _den_lh_ois = RWBit(_LSM6DS_UI_INT_OIS, 5)
+    _lvl2_ois = RWBit(_LSM6DS_UI_INT_OIS, 6)
+    _int2_drdy_ois = RWBit(_LSM6DS_UI_INT_OIS, 7)
+    _lpf_xl = RWBit(_LSM6DS_CTRL1_XL, 1)
+
+    _accel_range = RWBits(2, _LSM6DS_CTRL1_XL, 2)
+    _accel_data_rate = RWBits(4, _LSM6DS_CTRL1_XL, 4)
+
+    _gyro_data_rate = RWBits(4, _LSM6DS_CTRL2_G, 4)
+    _gyro_range = RWBits(2, _LSM6DS_CTRL2_G, 2)
+    _gyro_range_125dps = RWBit(_LSM6DS_CTRL2_G, 1)
+    _gyro_range_4000dps = RWBit(_LSM6DS_CTRL2_G, 0)
+
+    _sw_reset = RWBit(_LSM6DS_CTRL3_C, 0)
+    _sim = RWBit(_LSM6DS_CTRL3_C, 3)
+    _pp_od = RWBit(_LSM6DS_CTRL3_C, 4)
+    _h_lactive = RWBit(_LSM6DS_CTRL3_C, 5)
+    _bdu = RWBit(_LSM6DS_CTRL3_C, 6)
+    _boot = RWBit(_LSM6DS_CTRL3_C, 7)
+    _st_xl = RWBits(2, _LSM6DS_CTRL_5_C, 0)
+    _st_g = RWBits(2, _LSM6DS_CTRL_5_C, 2)
+    _rounding_status = RWBit(_LSM6DS_CTRL_5_C, 4)
+    _rounding = RWBits(2, _LSM6DS_CTRL_5_C, 5)
+    _xl_ulp_en = RWBit(_LSM6DS_CTRL_5_C, 7)
+    _aux_sens_on = RWBits(2, _LSM6DS_MASTER_CONFIG, 0)
+    _master_on = RWBit(_LSM6DS_MASTER_CONFIG, 2)
+    _shub_pu_en = RWBit(_LSM6DS_MASTER_CONFIG, 3)
+    _pass_through_mode = RWBit(_LSM6DS_MASTER_CONFIG, 4)
+    _start_config = RWBit(_LSM6DS_MASTER_CONFIG, 5)
+    _write_once = RWBit(_LSM6DS_MASTER_CONFIG, 6)
+    _rst_master_regs = RWBit(_LSM6DS_MASTER_CONFIG, 7)
+    _i3c_disable = RWBit(_LSM6DS_CTRL9_XL, 1)
+
+    _raw_temp = ROUnaryStruct(_LSM6DS_OUT_TEMP_L, "<h")
+
+    _raw_accel_data = Struct(_LSM6DS_OUTX_L_A, "<hhh")
+    _raw_gyro_data = Struct(_LSM6DS_OUTX_L_G, "<hhh")
+    CHIP_ID = None
+    def __init__(self, i2c_bus, address=_LSM6DS_DEFAULT_ADDRESS):
         self.i2c_device = i2c_device.I2CDevice(i2c_bus, address)
-
-        if self._chip_id not in [_LSM6DSOX_CHIP_ID, _ISM330DHCX_CHIP_ID]:
-            raise RuntimeError("Failed to find LSM6DSOX or ISM330DHCX - check your wiring!")
+        if self.CHIP_ID is None:
+            raise AttributeError("LSM6DS Parent Class cannot be directly instantiated")
+        if self._chip_id != self.CHIP_ID:
+            raise RuntimeError("Failed to find %s - check your wiring!"%self.__class__.__name__)
         self.reset()
 
         self._bdu = True
-        self._i3c_disable = True
-        self._if_inc = True
 
-        self._accel_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
-        self._gyro_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+        self.accelerometer_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+        self.gyro_data_rate = Rate.RATE_104_HZ #pylint: disable=no-member
+
+        self.accelerometer_range = AccelRange.RANGE_4G #pylint: disable=no-member
+        self.gyro_range = GyroRange.RANGE_250_DPS #pylint: disable=no-member
 
-        self._accel_range = AccelRange.RANGE_4G #pylint: disable=no-member
         self._cached_accel_range = self._accel_range
```

Comment:
why is this line here?

Reply:
I usually have lines like these to set the default values to something more useful, though 250dps  should change then as it's already the default.

Reply:
` self._cached_accel_range = self._accel_range`
should be handled within the property/setter right?

Reply:
yes, you're right. I misunderstood

Reply:
Pylint just reminded me that it doesn't like when you do that because it ends up getting defined outside of `__init__`
![image](https://user-images.githubusercontent.com/1183985/72403652-c1909c00-3707-11ea-853f-52dc74349933.png)

Reply:
set it to None at the top of __init__